### PR TITLE
init(ex/skate/snapshot_serde): create deserialize function

### PR DESCRIPTION
**Pull request by firestack**
Asana Ticket: https://app.asana.com/0/1203014709808707/1208675099752884/f

### CODE REVIEW

File: lib/skate/detours/detours.ex [hunk 1]
```diff
@@ -193,36 +193,33 @@ defmodule Skate.Detours.Detours do
...
   """
-  def update_or_create_detour_for_user(user_id, uuid, attrs \\ %{}) do
-    user = User.get_by_id!(user_id)
-
-    previous_record = uuid != nil && Skate.Repo.get(Detour, uuid)
-
-    detour =
-      case uuid do
-        nil ->
-          create_detour_for_user(user_id, attrs)
```
Comment (hannahpurcell) [inline lib/skate/detours/detours.ex:206]: Huh! I'm realizing that this case wasn't necessary, given the Repo.insert would handle that, right?

Comment (firestack) [reply lib/skate/detours/detours.ex:206]: Affirmative! I tried to figure it[^1] out when we first added this but ran into issues, but it went smoothly this time so 🤷🏻‍♀️ 

[^1]: "it" being `upsert` with a single call without a `case`

File: lib/skate_web/controllers/detours_controller.ex [hunk 2]
```diff
@@ -11,15 +11,11 @@ defmodule SkateWeb.DetoursController do
...
-  def update_snapshot(conn, %{"snapshot" => %{"context" => context} = snapshot}) do
-    uuid = Map.get(context, "uuid")
-
+  def update_snapshot(conn, %{"snapshot" => snapshot}) do
     %{id: user_id} = AuthManager.Plug.current_resource(conn)
 
     {:ok, %Skate.Detours.Db.Detour{id: returned_uuid}} =
-      Detours.update_or_create_detour_for_user(user_id, uuid, %{
-        state: snapshot
-      })
```
Comment (hannahpurcell) [inline lib/skate_web/controllers/detours_controller.ex:22]: I wonder why I previously wrapped the snapshot like this? I see that your changes assign state properly in the db, so... great! Much simpler 😊

File: lib/skate/detours/detours.ex (hunk 1 above)
Comment (hannahpurcell) [inline lib/skate/detours/detours.ex:206]: `create_detour_for_user` isn't being used anymore. What do you think: keep it or ditch it?

**Review by hannahpurcell — APPROVED** [review]
Looks great! The associated question is non-blocking

**Pull request merged**